src/backoff.js

```diff
@@ -1,28 +1,42 @@
-const apply = require('ramda/src/apply')
-const curry = require('ramda/src/curry')
+const apply  = require('ramda/src/apply')
+const curry  = require('ramda/src/curry')
+const ifElse = require('ramda/src/ifElse')
+const T      = require('ramda/src/T')
 
-// backoff :: Number -> Number -> (a... -> Promise b) -> a... -> Promise b
-const backoff = (base, tries, f) =>
-  (...args) => {
+const reject = Promise.reject.bind(Promise)
+
+// backoff :: { k: v } -> (a... -> Promise b) -> a... -> Promise b
+const backoff = (opts={}, f) => {
+  const {
+    base  = 250,
+    tries = Infinity,
```

Comment:
Is `Infinity` a safe default?

Reply:
I originally had something like `tries = 4` here, but then I couldn't think of a good reason to limit `tries` to any specific number by default.  `Infinity` is a valid number that means it will just keep trying until it succeeds.  If we want some other specific number for a particular reason, I'm happy to oblige.

Reply:
Seems to me that the safest course of action is to use some default limit of retries with the option to override with `Infinity` as necessary. Wouldn't want to accidentally flood a server with an "infinite" number of requests or something.

Besides, since the backoff is exponential, there will be a point where we've retried so many times & with such a long delay between attempts that it won't be helpful.

Just my opinion. Tis all quibbles.

Reply:
Would you settle for something on the order of `tries = 10`?